force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,136 @@
+public class OpportunityTriggerHandler extends TriggerHandler {
+    
+    private List<Opportunity> newOppList;
+    private List<Opportunity> oldOppList;
+    private Map<Id,Opportunity> newOppMap;
+    private Map<Id,Opportunity> oldOppMap;
+
+    public OpportunityTriggerHandler() {
+        
+        this.newOppList = (List<Opportunity>)Trigger.new;
+        this.oldOppList = (List<Opportunity>)Trigger.old;
+        this.newOppMap = (Map<Id,Opportunity>)Trigger.newMap;
+        this.oldOppMap = (Map<Id,Opportunity>)Trigger.oldMap;
+        //this.setMaxLoopCount(1);
+
+    }
+    
+    public override void beforeInsert() {
+
+        // Set default Type for new Opportunities            
+        for (Opportunity Opp : this.newOppList) {
+            if (opp.Type == null){
+                opp.Type = 'New Customer';
+            }
+        }        
+    }
```

Comment:
Logic if fine. See my comments on the account handler. Delegate this out to its own method.